### PR TITLE
Delta-1303 Boilerplate > FE > Update FE to fetch permissions and roles after Login

### PR DESCRIPTION
**Pull request by vcanonjr-freedom**
## Ticket references
- [Delta-1303](https://freedom.myjetbrains.com/youtrack/issue/Delta-1303)

## Summary of changes

Feature:
* Implemented feature to fetch permissions and roles after login

## Checklist

- [ ] bugfix
- [x] new feature
- [ ] breaking change
- [ ] added unit tests
- [ ] changes introduced has been discussed and approved
- [ ] requires manual testing
- [ ] documentation has been updated

## Testing

To test, checkout and run the BE branch of this [PR](https://github.com/anyTV/anytv-node-boilerplate/pull/116)
* Login and logout should work

### CODE REVIEW

File: src/mixins/transform-keys.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
+'use strict';
+
+import _ from 'lodash';
+
+function transformKeys(transformer) {
```
Comment (cedricabuso) [inline src/mixins/transform-keys.js:5]: @pandafox @jonelrempis @wildtron @wilbertverayin 
- I thought moving forward we will transition to snake case, so we should not add this in anymore?

Comment (pandafox) [reply src/mixins/transform-keys.js:5]: Depends on if we can consistently use snake casing throughout?

Comment (vcanonjr-freedom) [reply src/mixins/transform-keys.js]: I removed this for now

**Review by pandafox — APPROVED** [review]

**Review by jonelrempis — APPROVED** [review]

**Comment by kiko-f** [comment]
@vcanonjr-freedom after I enter my email and password then clicking Next button, an error message is displayed "Failed authentication. Please try again". On network the error displays "Cannot read property 'configured_messages' of undefined".

image

Screen Recording:
https://youtu.be/HjaTOsMLO1c

**Comment by kiko-f** [comment]
https://youtu.be/eztmL8pMJrc

**Review by johnurbano12 — COMMENTED** [review]
b3c4bb80-561a-42ab-bd1f-97afecc74cab
98094667-2b42-4495-b833-f62792a8b0df

**Review by johnurbano12 — APPROVED** [review]

**Review by cedricabuso — APPROVED** [review]

**Comment by jonelrempis** [comment]
- https://youtu.be/VN3JKR2qTA0

**Comment by wildtron** [comment]
* https://youtu.be/dnL9ok9fpYI

**Pull request merged**